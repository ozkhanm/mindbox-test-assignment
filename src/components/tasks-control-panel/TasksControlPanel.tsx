import React from "react";

import { FilterName } from "../../types";

interface TasksControlPanelProps {
    tasksNumber: number;
    activeFilterName: string;
    filterButtonClickHandler: (evt: React.ChangeEvent<HTMLInputElement>) => void;
    clearCompletedButtonClickHandler: () => void;
};

const TasksControlPanel: React.FC<TasksControlPanelProps> = ({tasksNumber, activeFilterName, filterButtonClickHandler, clearCompletedButtonClickHandler}) => {
    return (
        <div className="tasks-control-panel-wrapper">
            <p>{tasksNumber} items left</p>
            <div className="tasks-filter-controls">
                <input type="radio" id="radio-all" name="1" value="all" checked={activeFilterName === FilterName.ALL ? true : false} 
                    onChange={filterButtonClickHandler} />
                <label htmlFor="radio-all">All</label>    
                <input type="radio" id="radio-active" name="1" value="active" checked={activeFilterName === FilterName.ACTIVE ? true : false} 
                    onChange={filterButtonClickHandler} />
                <label htmlFor="radio-active">Active</label>
                <input type="radio" id="radio-completed" name="1" value="completed" checked={activeFilterName === FilterName.COMPLETED ? true : false}
                    onChange={filterButtonClickHandler} />
                <label htmlFor="radio-completed">Completed</label>
            </div>
            <div>
                <button className="clear-completed-button" type="button" onClick={clearCompletedButtonClickHandler}>Clear completed</button>
            </div>
        </div>
    );
};

export default TasksControlPanel;