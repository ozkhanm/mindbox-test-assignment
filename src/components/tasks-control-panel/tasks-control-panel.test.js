import React from "react";
import renderer from "react-test-renderer";

import TasksControlPanel from "./TasksControlPanel";

it(`PageHeader renders correctly`, () => {
    const tree = renderer
        .create(<TasksControlPanel />)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});
