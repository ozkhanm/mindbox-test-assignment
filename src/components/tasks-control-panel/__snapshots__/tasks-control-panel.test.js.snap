// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders correctly 1`] = `
<div
  className="tasks-control-panel-wrapper"
>
  <p>
     items left
  </p>
  <div
    className="tasks-filter-controls"
  >
    <input
      checked={false}
      id="radio-all"
      name="1"
      type="radio"
      value="all"
    />
    <label
      htmlFor="radio-all"
    >
      All
    </label>
    <input
      checked={false}
      id="radio-active"
      name="1"
      type="radio"
      value="active"
    />
    <label
      htmlFor="radio-active"
    >
      Active
    </label>
    <input
      checked={false}
      id="radio-completed"
      name="1"
      type="radio"
      value="completed"
    />
    <label
      htmlFor="radio-completed"
    >
      Completed
    </label>
  </div>
  <div>
    <button
      className="clear-completed-button"
      type="button"
    >
      Clear completed
    </button>
  </div>
</div>
`;
